### PR TITLE
Jest tests added for `.vue` components

### PR DESCRIPTION
**Pull request by OorfeneD**
#### What is done:
Added unit tests for **Vue** components:
- `Todo`
- `TodoList`
- `AddTodo`
- `Icon`
- `SignInModal`
- `SignUpModal`
- `index` page

#### What isn`t done yet:
Tests for **store**, **plugins** and **layouts**.

Good code coverage make it possible to find errors and mistakes before ones reaches production

### CODE REVIEW

**Comment by OorfeneD** [comment]
Due to #11 some test are needed to be changed

File: test/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      removeTodo: jest.fn(),
+      updateTodo: jest.fn()
+    }
+    getters = {
+      isLoggedIn: () => true
+    }
+    emptyGetters = {
+      isLoggedIn: () => false
+    }
+    store = new VueX.Store({
```
Comment (OorfeneD) [inline test/index.spec.js:32]: eslint needs to be disabled at this line

File: test/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      actions,
+      modules: {
+        user: {
+          getters,
+          namespaced: true
+        }
+      }
+    })
+    emptyStore = new VueX.Store({
```
Comment (OorfeneD) [inline test/index.spec.js:41]: eslint needs to be disabled at this line

File: test/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      wrapper.findComponent(AddTodo).vm.$emit('error', { msg: '' })
+      expect(spies.showError).toHaveBeenCalled()
+      wrapper.vm.$forceUpdate()
+    })
+    test('dispatches `removeTodo` action on `TodoList` component events', () => {
+      wrapper.findComponent(TodoList).vm.$emit('removetodo', { todo })
+      expect(actions.removeTodo).toHaveBeenCalled()
+    })
+    test('dispatches `updateTodo` action on `TodoList` component events', () => {
+      wrapper.findComponent(TodoList).vm.$emit('updatetodo', todo)
```
Comment (OorfeneD) [inline test/index.spec.js:94]: #11 break this

**Review by OorfeneD — COMMENTED** [review]
some fixes are required here due to #11 and Actions warnings

**Comment by Adequm** [comment]
Шиза

**Pull request merged**